internal/glfwkeyconversion/map-modifiers-to-macos.go

```diff
@@ -0,0 +1,52 @@
+package glfwkeyconversion
+
+import "github.com/go-gl/glfw/v3.3/glfw"
+
+// Modifier key masks pulled from flutter/packages/flutter/lib/src/services/raw_keyboard_macos.dart
+// URL: https://github.com/flutter/flutter/blob/3e63411256cc88afc48044aa5ea06c5c9c6a6846/packages/flutter/lib/src/services/raw_keyboard_macos.dart#L241
+const (
+	modifierControl    = 0x40000
+	modifierShift      = 0x20000
+	modifierOption     = 0x80000
+	modifierCommand    = 0x100000
+	modifierCapsLock   = 0x10000
+	modifierNumericPad = 0x200000
+)
+
+var modifierKeytoMods = map[glfw.Key]int{
+	glfw.KeyLeftControl:  modifierControl,
+	glfw.KeyLeftShift:    modifierShift,
+	glfw.KeyLeftAlt:      modifierOption,
+	glfw.KeyLeftSuper:    modifierCommand,
+	glfw.KeyRightControl: modifierControl,
+	glfw.KeyRightShift:   modifierShift,
+	glfw.KeyRightAlt:     modifierOption,
+	glfw.KeyRightSuper:   modifierCommand,
+	glfw.KeyCapsLock:     modifierCapsLock,
+	glfw.KeyNumLock:      modifierNumericPad,
+}
+
+// AsMacOSModifiers translate the keycode to the ModifierKey
+func AsMacOSModifiers(keycode glfw.Key) (int, bool) {
+	val, ok := modifierKeytoMods[keycode]
```

Comment:
a simple switch statement is likely more performant.

Reply:
It's an interesting topic. According to this blogpost the map is actually faster, but not much difference in absolute times. https://hashrocket.com/blog/posts/switch-vs-map-which-is-the-better-way-to-branch-in-go

Reply:
yeah, I ran across that which is why I reverted my thinking.

its actually an interesting optimization that could be done to the switch statements at the compiler level.

I suspect its because maps are hashing the key which is a fixed cost based on the key length. then doing a lookup. whereas a switch statement is doing multiple comparisons between the key and the case values.